tools/restatectl/src/commands/partition/list.rs

```diff
@@ -55,17 +76,24 @@ pub async fn list_partitions(
     let mut client =
         ClusterCtrlSvcClient::new(channel).accept_compressed(CompressionEncoding::Gzip);
 
-    let req = ClusterStateRequest::default();
-    let state = client
-        .get_cluster_state(req)
+    let cluster_state_request = ClusterStateRequest::default();
+    let cluster_state = client
+        .get_cluster_state(cluster_state_request)
         .await?
         .into_inner()
         .cluster_state
         .ok_or_else(|| anyhow::anyhow!("no cluster state returned"))?;
 
-    let mut partitions: BTreeMap<u32, Vec<PartitionListEntry>> = BTreeMap::new();
+    // we need the logs to show the current sequencer for each partition's log
+    let list_logs_request = ListLogsRequest::default();
+    let list_logs_response = client.list_logs(list_logs_request).await?.into_inner();
```

Comment:
Another improvement is to make these in parallel. I expect I'll come back to do some refactoring here though; currently there are some duplicated pieces of info loaded by the `status` command but it will require splitting all the list commands into data fetching (with requests fired in parallel) + list rendering pieces to make it work well. I wanted to get this functionality in sooner.

Reply:
Sounds like a nice improvement for the responsiveness. Can be a good follow-up. As you've said, it is not required though.